src/pf/anf.lua

```diff
@@ -247,7 +247,8 @@ local function renumber(expr)
    end
    local function lookup(var)
       if var == 'len' then return var end
-      return assert(substs[var], "unbound variable: "..var)
+      local subst = assert(substs[var], "unbound variable: "..var)
+      return subst
```

Comment:
The weird and surprising thing about `assert` is that it returns all arguments when the assertion does not trigger. Here's the manual snippet (emphasis mine):

> Calls error if the value of its argument v is false (i.e., nil or false); otherwise, **returns all its arguments.**